Sources/Host/POSIX/Platform.cpp

```diff
@@ -151,6 +151,16 @@ int Platform::OpenFile(std::string const &path, uint32_t flags, uint32_t mode) {
 
 bool Platform::CloseFile(int fd) { return (::close(fd) == 0); }
 
+ssize_t Platform::FileRead(int fd, std::string &buf, size_t count, size_t offset) {
+  buf.resize(count);
+
+  ssize_t nread = ::pread(fd, &buf[0], count, offset);
```

Comment:
Oh cause there's the `offset` argument and you don't want to change the offset in the file for multiple file reads I suppose.